### PR TITLE
MRG: Improve lighting in 3D viz

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR reworks lighting in 3d viz scenes:

master | PR
---------|-------
image | image
image | image
image | image
image | image
image | image

Related to https://github.com/mne-tools/mne-python/pull/6731#issuecomment-532625434

### CODE REVIEW

**Comment by agramfort** [comment]
is there a lighting result (eg from pysurfer) that you are aiming for?

**Comment by GuillaumeFavelier** [comment]
The goal is to reproduce the lighting method used by Mayavi. Should I maybe update the table to integrate it?

**Comment by larsoner** [comment]
Yes I have noticed that they are different. The right still doesn't quite look like Mayavi, though -- the contrast is too large. Maybe the ambient light level is higher in Mayavi?

**Comment by agramfort** [comment]
I agree it requires some adjustment for mayavi like figure.

>

**Comment by GuillaumeFavelier** [comment]
Hm... I checked the `ambient` and `diffuse` values and they are the same ([mayavi](https://github.com/enthought/mayavi/blob/master/tvtk/tools/mlab.py#L850-L851) against [pyvista](https://github.com/pyvista/pyvista/blob/master/pyvista/plotting/plotting.py#L1088)) :thinking: 

I'll keep looking

**Comment by larsoner** [comment]
Difference in light follows camera vs not? Default light locations?

**Comment by larsoner** [comment]
Maybe this?

https://github.com/nipy/PySurfer/blob/master/surfer/viz.py#L3496-L3502

**Comment by GuillaumeFavelier** [comment]
It's trickier than it seems. It's technically 3 lights (of type `CameraLight`) but the first one (of type `HeadLight`) is actually disabled.

**Comment by GuillaumeFavelier** [comment]
*Updated table after e23c884*

master | PR
---------|-------
image | image
image |  image
image |  image
image | image
image | image

**Comment by agramfort** [comment]
I like the way it looks !

**Comment by GuillaumeFavelier** [comment]
@larsoner I pushed 4df834b to integrate https://github.com/mne-tools/mne-python/pull/8182#issuecomment-684932447

File: mne/viz/backends/_pyvista.py [hunk 1]
```diff
@@ -243,6 +245,35 @@ def subplot(self, x, y):
...
+        lights = list(self.plotter.renderer.GetLights())
+        lights.pop(0)  # unused headlight
+        lights[0].SetPosition(_to_pos(45.0, -45.0))
+        lights[1].SetPosition(_to_pos(-30.0, 60.0))
+        lights[2].SetPosition(_to_pos(-30.0, -60.0))
+
+    def update_lighting(self):
+        # Inspired from Mayavi's version of Raymond Maple 3-lights illumination
+        lights = list(self.plotter.renderer.GetLights())
+        headlight = lights.pop(0)
```
Comment (larsoner) [inline mne/viz/backends/_pyvista.py:258]: I'm a bit concerned that there are two `.pop`'s going on. Is it worth adding the conditional:
```
assert len(lights) in (3, 4)
if len(lights) == 4:
    lights.pop(0)
```
both places?

Comment (larsoner) [reply mne/viz/backends/_pyvista.py:258]: ohhh wait you don't actually remove the light from VTK. Never mind!

File: mne/viz/backends/_pyvista.py [hunk 2]
```diff
@@ -646,6 +677,15 @@ def _rad2deg(rad):
     return rad * 180. / np.pi
 
 
+def _to_pos(elevation, azimuth):
+    theta = azimuth * np.pi / 180.0
+    phi = (90.0 - elevation) * np.pi / 180.0
+    x = np.sin(theta) * np.sin(phi)
+    y = np.cos(phi)
+    z = np.cos(theta) * np.sin(phi)
+    return x, y, z
```
Comment (larsoner) [inline mne/viz/backends/_pyvista.py:686]: This should probably use `mne.transforms.sph_to_cart`

**Comment by larsoner** [comment]
I'll merge this, let's take care of `sph_to_cart` deduplication in another PR when someone sees it and remembers to do it

**Comment by larsoner** [comment]
Thanks @GuillaumeFavelier !

**Pull request merged**